lib/executor.js

```diff
@@ -346,6 +346,8 @@ function setupMiddleware(app, instructions) {
   app.defineMiddlewarePhases(phases);
 
   middleware.forEach(function(data) {
+    var middlewareUnexcluded =  _.find(instructions.files.boot, function(o) { return o === data.sourceFile });
+    if (!middlewareUnexcluded){ return; }
```

Comment:
This is a wrong place where to filter middleware. All filtering must happen in the compiler phase, executor is only interpreting the instructions produced by the compiler.

> However, when we debugged it, we saw that the scriptExtensions was undefined at the moment of filtering out files based on extensions. We traced this back to fixFileExtensions not received the options.scriptExtensions.

If `fixFileExtenisons` is not receiving `options.scriptExtensions`, then the way to go is to fix the compiler to correctly pass `options.scriptExtensions` all the way to `fixFileExtensions` when working with the middleware.

Reply:
@bajtos I see, thank you for clarifying. I actually cannot replicate the issue on version 2.x.x. It looks like you guys are grabbing the extensions from options as of 2.27.x Closing for now. Thank you!

Reply:
I am glad the issue has been resolved for you 👍